jar-infer/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -142,6 +128,38 @@
     writeJarModel(cha, map_mtd_result);
     return map_str_result;
   }
+  /*
+   * Unpack JAR archive and return directory path
+   *
+   */
+  private static String extractJAR(String jarPath) {
+    Preconditions.checkArgument(jarPath.endsWith(".jar"), "invalid jar path!");
+    System.out.println("extracting " + jarPath + "...");
+    String jarDir = jarPath.substring(0, jarPath.lastIndexOf('.'));
+    try {
+      JarFile jar = new JarFile(jarPath);
+      Enumeration enumEntries = jar.entries();
+      while (enumEntries.hasMoreElements()) {
+        JarEntry file = (JarEntry) enumEntries.nextElement();
+        File f = new File(jarDir + File.separator + file.getName());
+        if (file.isDirectory()) {
+          continue;
+        }
+        f.getParentFile().mkdirs();
+        InputStream is = jar.getInputStream(file);
+        FileOutputStream fos = new FileOutputStream(f);
+        while (is.available() > 0) {
+          fos.write(is.read());
+        }
+        fos.close();
+        is.close();
+      }
+      jar.close();
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
I would re-throw as an `Error` here, otherwise you risk silently producing an empty directory. Also, maybe catch `IOException` or something more precise than `Exception`? (You don't want to say, catch an NPE in this block)